lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -43,6 +43,46 @@ def self.ancestors_of(element_or_scope)
 
     end
 
+    class CrossAssignment < ::ActiveRecord::Base
+
+      belongs_to :cross_parent, class_name: 'PolicyElement', foreign_key: :cross_parent_id
+      belongs_to :cross_child, class_name: 'PolicyElement', foreign_key: :cross_child_id
+
+      def self.transitive_closure?(ancestor, descendant)
+        descendants_of(ancestor).include?(descendant)
+      end
+
+      def self.descendants_of(element_or_scope)
```

Comment:
Argh, it hurts to see my lazy code copypasta'd. Not your fault, though, I'll work on moving this back into the hierarchical sql gem or some other suitable place.